server/src/main/java/keywhiz/inject/ServiceDataSourceModule.java

```diff
@@ -0,0 +1,40 @@
+package keywhiz.inject;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import io.dropwizard.db.DataSourceFactory;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.setup.Environment;
+import keywhiz.JooqHealthCheck;
+import keywhiz.KeywhizConfig;
+import keywhiz.service.config.Readonly;
+
+import static keywhiz.JooqHealthCheck.OnFailure.LOG_ONLY;
+import static keywhiz.JooqHealthCheck.OnFailure.RETURN_UNHEALTHY;
+
+class ServiceDataSourceModule extends AbstractModule {
+  @Provides @Singleton ManagedDataSource dataSource(Environment environment,
+      KeywhizConfig config) {
+    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "db-writable");
+    environment.lifecycle().manage(dataSource);
+
+    environment.healthChecks().register("db-read-write-health",
+        new JooqHealthCheck(dataSource, LOG_ONLY));
+
+    return dataSource;
+  }
+
+  @Provides @Singleton @Readonly ManagedDataSource readonlyDataSource(Environment environment,
+      KeywhizConfig config) {
+    DataSourceFactory dataSourceFactory = config.getReadonlyDataSourceFactory();
+    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "db-readonly");
+    environment.lifecycle().manage(dataSource);
+
+    environment.healthChecks().register("db-readonly-health",
+        new JooqHealthCheck(dataSource, RETURN_UNHEALTHY));
+
+    return dataSource;
+  }
```

Comment:
This module is responsible for the service's data source bindings. If we performed the lifecycle and health check logic inside KeywhizService.doRun(), this module would go away. I should note that, per Guice best practices, these kinds of actions (initialization, lifecycle management, etc.) should not be performed within modules anyway, they're just for side-effect-free wiring.